src/log/log.c

```diff
@@ -2161,16 +2161,17 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
         WT_SET_LSN(&end_lsn, lastlog, 0);
         WT_ERR(__wt_fs_directory_list_free(session, &logfiles, logcount));
     }
-    if (lsnp != NULL) {
+
+    if (start_lsnp != NULL) {
         /*
          * Offsets must be on allocation boundaries. An invalid LSN from a user should just return
          * WT_NOTFOUND. It is not an error. But if it is from recovery, we expect valid LSNs so give
          * more information about that.
          */
-        if (lsnp->l.offset % allocsize != 0) {
-            if (LF_ISSET(WT_LOGSCAN_RECOVER))
+        if (start_lsnp->l.offset % allocsize != 0) {
+            if (LF_ISSET(WT_LOGSCAN_RECOVER | WT_LOGSCAN_RECOVER_METADATA))
```

Comment:
I am confused by the addition of WT_LOGSCAN_RECOVER_METADATA

Reply:
`cherry-pick` pulled it in from 4.2.  Do we want to remove it?

Reply:
The original change WT-4908 never had backports requested on it. I'd say remove it?

Reply:
Removed.